contrib/go/src/python/pants/contrib/go/targets/go_local_source.py

```diff
@@ -60,3 +61,16 @@ def __init__(self, address=None, payload=None, sources=None, **kwargs):
   def import_path(self):
     """The import path as used in import statements in `.go` source files."""
     return self.local_import_path(self.target_base, self.address)
+
+  # From `go help test`, ignore files beginning with "_" or ".", but otherwise match the glob
+  # "*_test.go".
+  _test_file_regexp = re.compile(r'^[^_\.].*_test\.go')
+
+  @classmethod
+  def _is_test_file(cls, src_path):
+    base = os.path.basename(src_path)
+    return re.match(cls._test_file_regexp, base) is not None
+
+  @property
```

Comment:
Because this will be relatively costly, you might consider either making it a memoized property, or converting it into a method to the send the signal that it is "doing things".

Reply:
Oops! Was trying to avoid over-memoizing things. I think an `@memoized_property` is appropriate for now as I can't immediately see how this would be extended with extra arguments, and I think semantically its use as a property is reasonable enough (since I don't think we're actually doing any filesystem stuff by selecting `.sources_paths`, as that is already done?).

Reply:
Converted to `@memoized_property`!

Reply:
Yea, correct: no filesystem access. Just linear time regex matching.